src/api/app/models/project.rb

```diff
@@ -1419,24 +1419,24 @@ def build_succeeded?(repository = nil)
     states = {}
     repository_states = {}
 
-    br = Buildresult.find(project: name, view: 'summary')
+    br = Buildresult.find_hashed(project: name, view: 'summary')
     # no longer there?
-    return false unless br
+    return false if br.empty?
 
-    br.each('result') do |result|
-      if repository && result.value(:repository) == repository
+    br.elements('result') do |result|
```

Comment:
There isn't an `elements` method if this isn't ActiveXML, or? 🤔

Reply:
elements is a Xmlhash function

Reply:
Ah, right. Forgot that we don't get a "normal" hash here.